networkx/readwrite/json_graph/node_link.py

```diff
@@ -182,10 +179,18 @@ def node_link_graph(data, directed=False, multigraph=True, attrs=None):
         if not multigraph:
             edgedata = dict((make_str(k), v) for k, v in d.items()
                             if k != source and k != target)
-            graph.add_edge(mapping[src], mapping[tgt], **edgedata)
+            if (isinstance(src, list)):
+                src = tuple(src)
+            if (isinstance(tgt, list)):
+                tgt = tuple(tgt)
+            graph.add_edge(src, tgt, **edgedata)
         else:
             ky = d.get(key, None)
             edgedata = dict((make_str(k), v) for k, v in d.items()
                             if k != source and k != target and k != key)
-            graph.add_edge(mapping[src], mapping[tgt], ky, **edgedata)
+            if (isinstance(src, list)):
+                src = tuple(src)
+            if (isinstance(tgt, list)):
+                tgt = tuple(tgt)
```

Comment:
Instead of doing this twice you can do
``` python
src = tuple(d[source]) if isinstance(d[source], list) else d[source]       
tgt = tuple(d[target]) if isinstance(d[target], list) else d[target]
```
while accessing the values in the starting of the for loop